src/inmanta/deploy/tasks.py

```diff
@@ -45,13 +46,17 @@ class Task(abc.ABC):
     """
 
     resource: ResourceIdStr
-
     id: resources.Id = dataclasses.field(init=False, compare=False, hash=False)
+    reason: typing.Optional[str] = dataclasses.field(compare=False, hash=False, default=None)
```

Comment:
Adding this field here has some implications. The `AgentQueues` class, for example, uses this object as key in a dictionary. This means that it relies on the fact that the hash of the object is identifying for the object. So in the previous implementation every `Deploy()` object for the same resource_id would be considered equal (also with respect to the `__eq__()` method). Now that we add the reason field, that's no longer true. Deploy object will be considered different if they have a different reason. I don't think that's something we want.

Reply:
You're right, but given the attributes in the `dataclasses.field`, everything should work as expected.

Reply:
it might make sense to add this to `PrioritizedTask` instead of task, it resolve the issue of optionality as well.  (or even QueueItem)

Reply:
> You're right, but given the attributes in the dataclasses.field, everything should work as expected.

Indeed. I overlooked that.